src/main/kotlin/org/elm/ide/toolwindow/ElmCompilerPanel.kt

```diff
@@ -0,0 +1,298 @@
+package org.elm.ide.toolwindow
+
+import com.intellij.ide.CommonActionsManager
+import com.intellij.ide.OccurenceNavigator
+import com.intellij.ide.util.PsiNavigationSupport
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.actionSystem.ActionManager
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.actionSystem.DefaultActionGroup
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.fileEditor.OpenFileDescriptor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.SimpleToolWindowPanel
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.ui.AutoScrollToSourceHandler
+import com.intellij.ui.JBSplitter
+import com.intellij.ui.components.JBLabel
+import com.intellij.ui.components.JBScrollPane
+import com.intellij.ui.content.ContentManager
+import com.intellij.ui.table.JBTable
+import org.elm.openapiext.checkIsEventDispatchThread
+import org.elm.openapiext.toPsiFile
+import org.elm.workspace.ElmProject
+import org.elm.workspace.compiler.CompilerMessage
+import org.elm.workspace.compiler.ElmBuildAction
+import org.elm.workspace.compiler.Region
+import org.elm.workspace.compiler.Start
+import org.elm.workspace.elmWorkspace
+import java.awt.Color
+import java.awt.Component
+import java.awt.Dimension
+import java.awt.font.TextAttribute
+import java.util.*
+import javax.swing.JComponent
+import javax.swing.JTable
+import javax.swing.JTextPane
+import javax.swing.ListSelectionModel
+import javax.swing.ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER
+import javax.swing.ScrollPaneConstants.VERTICAL_SCROLLBAR_AS_NEEDED
+import javax.swing.border.EmptyBorder
+import javax.swing.table.DefaultTableCellRenderer
+import javax.swing.table.DefaultTableModel
+
+
+class ElmCompilerPanel(private val project: Project, private val contentManager: ContentManager) : SimpleToolWindowPanel(true, false), Disposable, OccurenceNavigator {
+
+    // TODO: support multiple projects ? how is 'elm make' supported for multiple projects anyway ?
+    lateinit var elmProjects: List<ElmProject>
+
+    private fun updateSelectionAndCreateOccurenceInfo(): OccurenceNavigator.OccurenceInfo {
+        // update selection
+        messageUI.text = compilerMessages[indexCompilerMessages].messageWithRegion.message
+        errorTableUI.setRowSelectionInterval(indexCompilerMessages, indexCompilerMessages)
+
+        // create occurence info
+        val (virtualFile, document, start) = startFromErrorMessage(VfsUtil.findFile(elmProjects[0].manifestPath, true))
```

Comment:
There is no guarantee that `elmProjects` is non-empty. So this may crash (`elmProjects[0]`).

Reply:
yep... that was a short minded fix